src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -52,93 +76,1173 @@ static ColliderCylinderInit D_80A41B00 = {
+            EnGo_SetupAction(this, func_80A40B1C);
+        } else {
+            EnGo_SetupAction(this, EnGo_BiggoronActionFunc);
+        }
+    }
+}
+
+void EnGo_BiggoronActionFunc(EnGo* this, GlobalContext* globalCtx) {
+    if (((this->actor.params & 0xF0) == 0x90) && (this->unk_1E0.unk_00 == 2)) {
+        if (gSaveContext.bgsFlag) {
+            this->unk_1E0.unk_00 = 0;
+        } else {
+            if (INV_CONTENT(ITEM_TRADE_ADULT) == ITEM_EYEDROPS) {
+                EnGo_ChangeAnimation(this, 2);
+                this->unk_21E = 100;
+                this->unk_1E0.unk_00 = 0;
+                EnGo_SetupAction(this, EnGo_Eyedrops);
+                globalCtx->msgCtx.msgMode = 0x37;
+                gSaveContext.timer2State = 0;
+                func_800800F8(globalCtx, 0x105E, -0x63, &this->actor, 0);
+            } else {
+                this->unk_1E0.unk_00 = 0;
+                EnGo_SetupAction(this, EnGo_GetItem);
+                func_80106CCC(globalCtx);
+                EnGo_GetItem(this, globalCtx);
+            }
+        }
+    } else if (((this->actor.params & 0xF0) == 0) && (this->unk_1E0.unk_00 == 2)) {
+        EnGo_SetupAction(this, EnGo_GetItem);
+        globalCtx->msgCtx.unk_E3E7 = 4;
+        globalCtx->msgCtx.msgMode = 0x36;
+    } else {
+        if ((DECR(this->unk_212) == 0) && !EnGo_IsCameraModified(this, globalCtx)) {
+            EnGo_ReverseAnimation(this);
+            this->skelAnime.playSpeed = -0.1f;
+            this->skelAnime.playSpeed *= (this->actor.params & 0xF0) == 0x90 ? 0.5f : 1.0f;
+            EnGo_SetupAction(this, func_80A408D8);
+        }
+    }
+}
+
+void func_80A408D8(EnGo* this, GlobalContext* globalCtx) {
+    f32 frame;
+
+    if (this->skelAnime.playSpeed != 0.0f) {
+        Math_SmoothStepToF(&this->skelAnime.playSpeed, ((this->actor.params & 0xF0) == 0x90 ? 0.5f : 1.0f) * -1.0f,
+                           0.1f, 1000.0f, 0.1f);
+        frame = this->skelAnime.curFrame;
+        frame += this->skelAnime.playSpeed;
+        if (frame >= 12.0f) {
+            return;
+        } else {
+            this->skelAnime.curFrame = 12.0f;
+            this->skelAnime.playSpeed = 0.0f;
+            if ((this->actor.params & 0xF0) != 0x90) {
+                this->unk_212 = 30;
+                return;
+            }
+        }
+    }
+
+    if (DECR(this->unk_212) == 0) {
+        EnGo_SetupAction(this, func_80A40494);
+    } else if (EnGo_IsCameraModified(this, globalCtx)) {
+        EnGo_ReverseAnimation(this);
+        Audio_PlaySoundGeneral(NA_SE_EN_GOLON_SIT_DOWN, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+        this->skelAnime.playSpeed = 0.0f;
+        EnGo_SetupAction(this, func_80A405CC);
+    }
+}
+
+void func_80A40A54(EnGo* this, GlobalContext* globalCtx) {
+    f32 float1 = ((f32)0X8000 / Animation_GetLastFrame(&D_06010590));
```

Comment:
Looks like there are a few other hex numbers to fix to 0x in these files, including here:
```suggestion
    f32 float1 = ((f32)0x8000 / Animation_GetLastFrame(&D_06010590));
```